easy-random-core/src/main/java/org/jeasy/random/FieldPopulator.java

```diff
@@ -128,7 +127,7 @@ private Object generateRandomValue(final Field field, final RandomizationContext
             value = mapPopulator.getRandomMap(field, context);
         } else {
             if (context.getParameters().isScanClasspathForConcreteTypes() && isAbstract(fieldType) && !isEnumType(fieldType) /*enums can be abstract, but can not inherit*/) {
-                Class<?> randomConcreteSubType = randomElementOf(filterSameParameterizedTypes(getPublicConcreteSubTypesOf(fieldType), fieldGenericType));
+                Class<?> randomConcreteSubType = context.randomElementOf(filterSameParameterizedTypes(getPublicConcreteSubTypesOf(fieldType), fieldGenericType));
```

Comment:
FieldPopulator has a reference to an EasyRandom instance (which is initialized with the same seed), so this can be replaced with:

```java
List<Class<?>> parameterizedTypes = filterSameParameterizedTypes(getPublicConcreteSubTypesOf(fieldType), fieldGenericType);
if (parameterizedTypes.size() == 0) {
    throw new ObjectCreationException("Unable to find a matching concrete subtype of type: " + fieldType);
} else {
    Class<?> randomConcreteSubType = parameterizedTypes.get(easyRandom.nextInt(parameterizedTypes.size()));
    value = easyRandom.doPopulateBean(randomConcreteSubType, context);
}
```

Reply:
thanks for exact code sample. done